x/pairing/keeper/unresponsive_provider_test.go

```diff
@@ -0,0 +1,331 @@
+	}
+
+	// send the relay requests (provider gets payment)
+	_, err = ts.servers.PairingServer.RelayPayment(ts.ctx, &types.MsgRelayPayment{Creator: ts.providers[0].address.String(), Relays: Relays})
+	require.Nil(t, err)
+
+	// advance enough epochs so the unresponsive provider will be punished
+	if largerConst < recommendedEpochNumToCollectPayment {
+		largerConst = recommendedEpochNumToCollectPayment
+	}
+	for i := uint64(0); i < largerConst; i++ {
+		ts.ctx = testkeeper.AdvanceEpoch(ts.ctx, ts.keepers)
+	}
+
+	// test the provider has been unstaked
+	_, unStakeStoragefound, _ := ts.keepers.Epochstorage.UnstakeEntryByAddress(sdk.UnwrapSDKContext(ts.ctx), epochstoragetypes.ProviderKey, ts.providers[1].address)
+	require.True(t, unStakeStoragefound)
+	_, stakeStorageFound, _ := ts.keepers.Epochstorage.GetStakeEntryByAddressCurrent(sdk.UnwrapSDKContext(ts.ctx), epochstoragetypes.ProviderKey, ts.spec.Name, ts.providers[1].address)
+	require.False(t, stakeStorageFound)
+
+	// validate the complainers CU field in provider1's providerPaymentStorage has been reset after being punished (note we use the epoch from the relay because that is when it got reported)
+	providerPaymentStorageKey := ts.keepers.Pairing.GetProviderPaymentStorageKey(sdk.UnwrapSDKContext(ts.ctx), ts.spec.Name, uint64(relayEpoch), ts.providers[1].address)
+	providerPaymentStorage, found := ts.keepers.Pairing.GetProviderPaymentStorage(sdk.UnwrapSDKContext(ts.ctx), providerPaymentStorageKey)
+	require.Equal(t, true, found)
+	require.Equal(t, uint64(0), providerPaymentStorage.GetComplainersTotalCu())
+
+	// advance some more epochs
+	for i := 0; i < 2; i++ {
+		ts.ctx = testkeeper.AdvanceEpoch(ts.ctx, ts.keepers)
+	}
+
+	// create more relay requests for provider0 that contain complaints about provider1 (note, sessionID changed)
+	var RelaysAfter []*types.RelayRequest
+	for clientIndex := 0; clientIndex < testClientAmount; clientIndex++ { // testing testClientAmount of complaints
+
+		relayRequest := &types.RelayRequest{
+			Provider:              ts.providers[0].address.String(),
+			ApiUrl:                "",
+			Data:                  []byte(ts.spec.Apis[0].Name),
+			SessionId:             uint64(2),
+			ChainID:               ts.spec.Name,
+			CuSum:                 ts.spec.Apis[0].ComputeUnits * 10,
+			BlockHeight:           sdk.UnwrapSDKContext(ts.ctx).BlockHeight(),
+			RelayNum:              0,
+			RequestBlock:          -1,
+			DataReliability:       nil,
+			UnresponsiveProviders: unresponsiveProvidersData, // create the complaint
+		}
+		sig, err := sigs.SignRelay(ts.clients[clientIndex].secretKey, *relayRequest)
+		relayRequest.Sig = sig
+		require.Nil(t, err)
+		RelaysAfter = append(RelaysAfter, relayRequest)
+	}
+
+	// send the relay requests (provider gets payment)
+	_, err = ts.servers.PairingServer.RelayPayment(ts.ctx, &types.MsgRelayPayment{Creator: ts.providers[0].address.String(), Relays: RelaysAfter})
+	require.Nil(t, err)
+
```

Comment:
let's verify that even though the provider isn't punishable, other providers can still claim money regardless, meaning let's test that errors unstaking doesn't hinder the relay payment validity

Reply:
when I wrote the relay payment unit tests, I've written a helper function that performs payment and checks balances, so I switched every relayPayment call with my helper function